src/helper-modules/ordnance-survey/ordnance-survey.service.test.ts

```diff
@@ -0,0 +1,126 @@
+import { HttpService } from '@nestjs/axios';
+import { ConfigService } from '@nestjs/config';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { AxiosError } from 'axios';
+import { when } from 'jest-when';
+import { of, throwError } from 'rxjs';
+import expectedResponse = require('./examples/example-response-for-search-places-v1-postcode.json');
+import noResultsResponse = require('./examples/example-response-for-search-places-v1-postcode-no-results.json');
+
+import { GEOSPATIAL } from '@ukef/constants';
+
+import { OrdnanceSurveyException } from './exception/ordnance-survey.exception';
+import { OrdnanceSurveyService } from './ordnance-survey.service';
+
+describe('OrdnanceSurveyService', () => {
+  const valueGenerator = new RandomValueGenerator();
+
+  let httpServiceGet: jest.Mock;
+  let configServiceGet: jest.Mock;
+  let service: OrdnanceSurveyService;
+
+  const testPostcode = GEOSPATIAL.EXAMPLES.POSTCODE;
+  const testKey = valueGenerator.string({ length: 10 });
+  const basePath = '/search/places/v1/postcode';
+
+  beforeEach(() => {
+    const httpService = new HttpService();
+    const configService = new ConfigService();
+    httpServiceGet = jest.fn();
+    httpService.get = httpServiceGet;
+
+    configServiceGet = jest.fn().mockReturnValue({ key: testKey });
+    configService.get = configServiceGet;
+
+    service = new OrdnanceSurveyService(httpService, configService);
+  });
+
+  describe('getAddressesByPostcode', () => {
+    const expectedPath = `${basePath}?postcode=${encodeURIComponent(testPostcode)}&lr=EN&key=${encodeURIComponent(testKey)}`;
+
+    const expectedHttpServiceGetArgs: [string, object] = [expectedPath, { headers: { 'Content-Type': 'application/json' } }];
+
+    it('sends a GET to the Ordnance Survey API /search endpoint with the specified request', async () => {
+      when(httpServiceGet)
+        .calledWith(...expectedHttpServiceGetArgs)
+        .mockReturnValueOnce(
+          of({
+            data: expectedResponse,
+            status: 200,
+            statusText: 'OK',
+            config: undefined,
+            headers: undefined,
+          }),
+        );
+
+      await service.getAddressesByPostcode(testPostcode);
+
+      expect(httpServiceGet).toHaveBeenCalledTimes(1);
+      expect(httpServiceGet).toHaveBeenCalledWith(...expectedHttpServiceGetArgs);
+    });
+
+    it.each([
+      {
+        postcode: 'W1A 1AA',
+        expectedUrlQueryPart: `?postcode=W1A%201AA`,
+      },
+      {
+        postcode: 'W1A1AA',
+        expectedUrlQueryPart: '?postcode=W1A1AA',
+      },
+    ])('call Ordnance Survey API with correct and safe query parameters "$expectedUrlQueryPart"', async ({ postcode, expectedUrlQueryPart }) => {
```

Comment:
```suggestion
    ])('calls Ordnance Survey API with correct and safe query parameters "$expectedUrlQueryPart"', async ({ postcode, expectedUrlQueryPart }) => {
```

Reply:
Merged 1+2 tests to have 3 cases in describe.each, see comment bellow.